libra-node/src/main_node.rs

```diff
@@ -298,6 +300,21 @@ pub fn setup_environment(node_config: &mut NodeConfig) -> LibraHandle {
             node_config.base.waypoint.expect("No waypoint in config"),
         );
 
+        // Set up to listen for network configuration changes from StateSync.
+        match network_builder.conn_mgr_reqs_tx() {
```

Comment:
let's use an expect, because otherwise we're in a really bad state.

Reply:
honestly the fact that we even have a "NetworkBuilder" without a connectivity manager seems ... wrong.

Reply:
> honestly the fact that we even have a "NetworkBuilder" without a connectivity manager seems ... wrong.

I can't say that I disagree, but it is what it is.  Question arises later as to why the connectivity manager is tied to an authenticated network.

Reply:
Where exactly do you want the expect heree?